test/parallel/test-fs-cp.mjs

```diff
@@ -746,24 +746,26 @@ if (!isWindows) {
      }));
 }
 
-// Copy async should not throw exception if child folder is filtered out.
+// Copy should not throw exception if child folder is filtered out.
 {
   const src = nextdir();
-  mkdirSync(join(src, 'test-cp-async'), mustNotMutateObjectDeep({ recursive: true }));
+  mkdirSync(join(src, 'test-cp'), mustNotMutateObjectDeep({ recursive: true }));
 
   const dest = nextdir();
   mkdirSync(dest, mustNotMutateObjectDeep({ recursive: true }));
-  writeFileSync(join(dest, 'test-cp-async'), 'test-content', mustNotMutateObjectDeep({ mode: 0o444 }));
+  writeFileSync(join(dest, 'test-cp'), 'test-content', mustNotMutateObjectDeep({ mode: 0o444 }));
 
-  cp(src, dest, {
-    filter: (path) => !path.includes('test-cp-async'),
+  const opts = {
+    filter: (path) => !path.includes('test-cp'),
     recursive: true,
-  }, mustCall((err) => {
+  };
+  cp(src, dest, opts, mustCall((err) => {
     assert.strictEqual(err, null);
   }));
+  cpSync(src, dest, mustNotMutateObjectDeep(opts));
```

Comment:
Before fixing, this line throw following error:
```
SystemError [ERR_FS_CP_DIR_TO_NON_DIR]: Cannot overwrite directory with non-directory: cp returned EISDIR (cannot overwrite directory /Users/xxx/dev/node/test/.tmp.0/copy_65/test-cp with non-directory /Users/tho/dev/node/test/.tmp.0/copy_66/test-cp) /Users/tho/dev/node/test/.tmp.0/copy_66/test-cp
    at new SystemError (node:internal/errors:244:5)
    at new NodeError (node:internal/errors:355:7)
    at checkPathsSync (node:internal/fs/cp/cp-sync:77:13)
    at copyDir (node:internal/fs/cp/cp-sync:287:28)
    at onDir (node:internal/fs/cp/cp-sync:268:10)
    at getStats (node:internal/fs/cp/cp-sync:171:12)
    at handleFilterAndCopy (node:internal/fs/cp/cp-sync:158:10)
    at cpSyncFn (node:internal/fs/cp/cp-sync:60:10)
    at cpSync (node:fs:2892:3)
    at file:///Users/tho/dev/node/test/parallel/test-fs-cp.mjs:762:3 {
  code: 'ERR_FS_CP_DIR_TO_NON_DIR',
  info: {
    message: 'cannot overwrite directory /Users/xxx/dev/node/test/.tmp.0/copy_65/test-cp with non-directory /Users/xxx/dev/node/test/.tmp.0/copy_66/test-cp',
    path: '/Users/xxx/dev/node/test/.tmp.0/copy_66/test-cp',
    syscall: 'cp',
    errno: 21,
    code: 'EISDIR'
  },
  errno: [Getter/Setter],
  syscall: [Getter/Setter],
  path: [Getter/Setter]
}
```

Reply:
That should probably be fixed in a separate PR, let's not use `mustNotMutateObjectDeep` for now.

Reply:
@aduh95 `mustNotMutateObjectDeep` has been removed